migration-engine/core/src/migration/datamodel_migration_steps_inferrer.rs

```diff
@@ -230,6 +234,66 @@ impl<'a> DataModelMigrationStepsInferrerImpl<'a> {
         result
     }
 
+    fn indexes_to_create(&self) -> Vec<CreateIndex> {
+        let mut result = Vec::new();
+        for next_model in self.next.models() {
+            let previous_model = self.previous.find_model(&next_model.name);
+
+            for next_index in &next_model.indexes {
+                let index_present = previous_model
+                    .and_then(|previous_model| {
+                        previous_model
+                            .indexes
+                            .iter()
+                            .find(|previous_index| *previous_index == next_index)
+                    })
+                    .is_some();
+                if !index_present {
+                    result.push(CreateIndex {
+                        model: next_model.name.clone(),
+                        name: next_index.name.clone(),
+                        is_unique: next_index.is_unique,
+                        fields: next_index.fields.clone(),
+                    })
+                }
+            }
+        }
+
+        result
+    }
+
+    fn indexes_to_delete(&self) -> Vec<DeleteIndex> {
+        self.previous
+            .models()
+            .filter_map(|previous_model| {
+                self.next
+                    .models()
+                    .find(|next_model| next_model.name == previous_model.name)
+                    .map(|next_model| (previous_model, next_model))
+            })
+            .flat_map(|(previous_model, next_model)| {
+                previous_model.indexes.iter().filter_map(move |existing_index| {
+                    let still_exists = next_model
+                        .indexes
+                        .iter()
+                        .find(|new_index| *new_index == existing_index)
+                        .is_some();
```

Comment:
This check looks very similar to the one from above. I wonder if we should factor out this logic into a function either on `Model` directly or on the `ModelExtensions` trait.

Reply:
Fixed in 3eaa04a29ccef13e71fa86336779eaffae2793db